### PR TITLE
client: Add WebSocketClient class

### PR DESCRIPTION
**Pull request by xDavidos**

### CODE REVIEW

File: client/src/classes/WebSocketClient.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    if(ctx.GetArgType(0) == js::Type::STRING)
+    {
+        std::string msg;
+        if(!ctx.GetArg(0, msg)) return;
+
+        ctx.Return(websocket->Send(msg));
+    }
+    else if (ctx.GetArgType(0) == js::Type::TYPED_ARRAY)
+    {
+        js::Buffer* buffer;
```
[the remark this answers is not in the document]
Comment (LeonMrBonnie) [reply client/src/classes/WebSocketClient.cpp:20]: I will fix this myself manually

File: client/src/factories/WebSocketClientFactory.cpp [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#include "FactoryHandler.h"
+#include "interfaces/IResource.h"
+
+// clang-format off
+static js::FactoryHandler webSocketClientCFactory(alt::IBaseObject::Type::RML_DOCUMENT, [](js::Object& args) -> alt::IBaseObject* {
```
Comment (LeonMrBonnie) [inline client/src/factories/WebSocketClientFactory.cpp:5]: Wrong type specified here

File: client/src/classes/WebSocketClient.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    tpl.Method<&alt::IWebSocketClient::Start>("start");
+    tpl.Method<&alt::IWebSocketClient::Stop>("stop");
+    tpl.Method("send", Send);
+    tpl.Method<&alt::IWebSocketClient::AddSubProtocol>("addSubProtocol");
+    tpl.Method<&alt::IWebSocketClient::GetSubProtocols>("GetSubProtocols");
+    tpl.Method<&alt::IWebSocketClient::SetExtraHeader>("setExtraHeader");
+    tpl.Method<&alt::IWebSocketClient::GetExtraHeaders>("getExtraHeaders");
+
+
```
Comment (LeonMrBonnie) [inline client/src/classes/WebSocketClient.cpp:53]: Stray empty line

**Comment by xDavidos** [comment]
If property use GetExtraInternalFieldValue we shoud use manualy binding?

**Comment by LeonMrBonnie** [comment]
> If property use GetExtraInternalFieldValue we shoud use manualy binding?

Yes, the automatic wrappers only work for alt:V entities.

**Pull request merged**